cmd/adapter/adapter.go

```diff
@@ -0,0 +1,430 @@
+	defer gzWriter.Close()
+	tarWriter := tar.NewWriter(gzWriter)
+	defer tarWriter.Close()
+
+	lql := buildLql("", ad.cfg.Logrange.Partition, dlLimitLines, -dlLimitLines)
+	ad.logger.Info("download(): Query=", lql)
+
+	var (
+		fnCntr = 0
+		err    error
+	)
+
+	buf := bytes.Buffer{}
+	err = api.Select(ctx, ad.lCli, &api.QueryRequest{Query: lql,
+		Limit: dlLimitLines}, false, func(res *api.QueryResult) {
+		if err == nil {
+			marshal(res.Events, &buf)
+			if buf.Len() > dlLimitPerFileBt {
+				err = writeTar(fnCntr, buf.Bytes(), tarWriter)
+				buf.Reset()
+				fnCntr++
+			}
+		}
+	})
+
+	if err == nil {
+		if buf.Len() > 0 {
+			err = writeTar(fnCntr, buf.Bytes(), tarWriter)
+		}
+	}
+
+	if err != nil {
+		ad.logger.Error("download(): Err=", err)
+		_ = writeTar(fnCntr, []byte("oops, something went wrong!\n"), tarWriter)
+	}
+
+	rw.Header().Set("Content-Disposition", "attachment; filename=logs.tar.gz")
+	return nil
+}
+
+func (ad *Adapter) makeHandlerWithCtx(ctx context.Context, handler handlerWithCtx) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		_ = handler(ctx, w, r, p)
+	}
+}
+
+//===================== adapter.cron =====================
+
+func (ad *Adapter) runCronQueries(ctx context.Context) {
+	if ad.cfg == nil || ad.cfg.Logrange == nil {
+		ad.logger.Warn("No config or Logrange config, skipping running cron queries")
+		return
+	}
+
+	ad.logger.Info("Running ", len(ad.cfg.Logrange.CronQueries), " cron queries")
+	for _, cq := range ad.cfg.Logrange.CronQueries {
+		ad.waitWg.Add(1)
+		go func(cq cronQuery) {
+			defer ad.waitWg.Done()
+			ad.runCronQuery(ctx, cq)
+		}(cq)
+	}
+}
+
+func (ad *Adapter) runCronQuery(ctx context.Context, cq cronQuery) {
+	ad.logger.Info("Entering into the loop to run \"", cq.Query, "\" every ", cq.PeriodSec, " seconds.")
+	for {
+		sleepDelay := time.Second * time.Duration(cq.PeriodSec)
+		res, err := ad.lCli.Execute(ctx, api.ExecRequest{Query: cq.Query})
+		if err != nil {
+			ad.logger.Warn("Could not connect to the server to run ", cq)
+			sleepDelay = time.Second
+		}
+
+		if res.Err != nil {
+			ad.logger.Error("Server returned error on executing ", cq.Query, " the err=", res.Err)
+		}
+
+		select {
+		case <-ctx.Done():
+			ad.logger.Info("Breaking up the runCronQuery loop for ", cq, " the context is closed.")
+			return
+		case <-time.After(sleepDelay):
+		}
+	}
+}
+
+//===================== adapter.sync =====================
+
+func (ad *Adapter) runSync(ctx context.Context) {
+	ad.logger.Info("Running sync every ", ad.cfg.SyncIntervalSec, " seconds...")
+	ticker := time.NewTicker(time.Second *
+		time.Duration(ad.cfg.SyncIntervalSec))
+
+	ad.waitWg.Add(1)
+	go func() {
+		for utils.Wait(ctx, ticker) {
+			ad.sync(ctx)
+		}
+		ad.logger.Warn("Sync stopped.")
+		ad.waitWg.Done()
+	}()
+}
+
+func (ad *Adapter) sync(ctx context.Context) {
+	ad.logger.Debug("sync(): Getting Logrange forwarder config...")
+	lrFwdCfg, err := ad.kCli.getLrFwdCfg()
```

Comment:
`getLrPwdCfg` -> `getForwarderConfig`
`getGrFwdCfg` -> `getGravityForwarderConfig`
No need to abbreviate when it hurts readability.

Reply:
applied (not yet merged to this branch)